granulate_utils/config_feeder/core/models/aggregation.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel, root_validator
+
+from granulate_utils.config_feeder.core.models.yarn import NodeYarnConfig, NodeYarnConfigCreate
+
+
+class CreateNodeConfigsRequest(BaseModel):
+    yarn_config: Optional[NodeYarnConfigCreate] = None
+
+    @root_validator
+    def at_least_one_config_is_required(cls, values: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
I don't understand what's going on here, the `yarn_config` can be set without `Optional` and default value thus pydantic will raise an exception upon creation. Also, I don't understand the `at_least_one`, the `yarn_config` can be either available or not, there's no list of configs

Reply:
Also, why don't you use a specific field validator?

Reply:
I will add autoscaling in next PR

```python
class CreateNodeConfigsRequest(BaseModel):
    yarn_config: Optional[NodeYarnConfigCreate] = None
    autoscaling_config: Optional[ClusterAutoScalingConfigCreate] = None

    @root_validator
    def at_least_one_config_is_required(cls, values: Dict[str, Any]) -> Dict[str, Any]:
        if values.get("yarn_config") is not None:
            return values
        if values.get("autoscaling_config") is not None:
            return values
        raise ValueError("at least one config is required")

```

Reply:
Ok